jpa/org.eclipse.persistence.jpa/src/org/eclipse/persistence/internal/jpa/QueryHintsHandler.java

```diff
@@ -355,6 +358,7 @@ static DatabaseQuery apply(String hintName, boolean shouldUseDefault, Object hin
                 query.setProperty(QUERY_HINT_PROPERTY, existingHints);
             }
             existingHints.put(hintName, hintValue);
+            query.setHintString(hintName);
```

Comment:
This is not completely wrong statement on the wrong place - why are you changing behaviour for all hints when Cursor hint is the one affected by this bug? Wouldn't 'CursorHint.applyToDatabaseQuery' be better place for this change? Why is 'query.setHintString(..)' call better then direct 'query.setIsPrepared(false)'?

Reply:
If there is any Hint set, this is were we are setting it to true or false, but we are not setting the name of the hint including cursors.  (with my limited knowledge)setIsPrepared(false) may trigger to prepare the stmt again and may take one more cycle.

Reply:
Are you sure that Hints can have only boolean values? Furthermore the method calls 'setIsPrepared(false)' under the hood - that is the reason, why new test is passing. Have you checked what is the purpose of the  'query.setHintString(..)' method, its javadoc/what this method does and how exactly it is implemented?

According to the [javadoc](https://github.com/eclipse-ee4j/eclipselink/blob/master/foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/queries/DatabaseQuery.java#L2152-L2174), the method is not there to set the name of the hint but to provide a hint to the database through additional comment in generated SQL statement.